### PR TITLE
Change RIOT's main loop to match Zephyr's style

### PR DESCRIPTION
**Pull request by ibriano**
And a little fix for the socket implementation.

### CODE REVIEW

**Comment by cabelitos** [comment]
+1

File: src/lib/comms/sol-socket-impl-riot.c [hunk 1]
```diff
@@ -349,6 +349,7 @@ sol_socket_ip_default_new(const struct sol_socket_options *options)
     socket->entry.pid = KERNEL_PID_UNDEF;
     socket->on_can_read = options->on_can_read;
     socket->on_can_write = options->on_can_write;
+    socket->data = options->data;
```
Comment (ceolin) [inline src/lib/comms/sol-socket-impl-riot.c]: :+1:

File: src/lib/comms/sol-socket-impl-riot.c [hunk 2]
```diff
@@ -349,6 +349,7 @@ sol_socket_ip_default_new(const struct sol_socket_options *options)
...
```
Comment (glima) [inline src/lib/comms/sol-socket-impl-riot.c]: This on Zephyr seems to fix its issue as well

**Comment by ibriano** [comment]
Merged

**Pull request merged**